sierra/src/main/java/org/httprpc/sierra/Picker.java

```diff
@@ -112,10 +114,28 @@ protected void processFocusEvent(FocusEvent event) {
 
         switch (event.getID()) {
             case FocusEvent.FOCUS_GAINED -> showPopup();
-            case FocusEvent.FOCUS_LOST -> hidePopup();
-        }
-    }
-
+            case FocusEvent.FOCUS_LOST -> {
+				// hide the popup, if focus did not move to it
+				if (!inPopup(event.getOppositeComponent())) {
+					hidePopup();
+				}
+				else {
+					requestFocus();
+				}
+			}
+		}
+	}
+
+	private boolean inPopup(Component component) {
```

Comment:
No need for `private` here because the class itself is already private.

Reply:
Picker class is abstract but not private... Shall I remove private?

Reply:
No, thank you for catching that. I hadn't looked closely enough and thought it was part of an inner class.